Framework/NexusGeometry/inc/MantidNexusGeometry/JSONInstrumentBuilder.h

```diff
@@ -0,0 +1,40 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_NEXUSGEOMETRY_JSONINSTRUMENTBUILDER_H_
+#define MANTID_NEXUSGEOMETRY_JSONINSTRUMENTBUILDER_H_
+
+#include "MantidGeometry/Instrument_fwd.h"
+#include "MantidNexusGeometry/DllConfig.h"
+#include "MantidNexusGeometry/JSONGeometryParser.h"
```

Comment:
You might be able to refactor the header such that this can be forward declared, such as storing by `unique_ptr`.

Reply:
Doesn't work because of Json::Value. I previously had issues trying to forward declare this type (even using `std::unique_ptr`). I got a `can't delete an incomplete type` error which has resurfaced when I try to forward declare `JSONGeometryParser`

Reply:
That's odd, I don't understand why that wouldn't be possible.